sdk/clientcore/core/src/main/java/io/clientcore/core/serialization/xml/XmlSerializable.java

```diff
@@ -0,0 +1,186 @@
+     *     &#47;&#47;
+     *     &#47;&#47; The following is the equivalent of:
+     *     &#47;&#47; - XmlReader.currentToken&#40;&#41; == XmlToken.START_ELEMENT
+     *     &#47;&#47; - XmlReader.getElementName&#40;&#41;.getNamespaceURI&#40;&#41;.equals&#40;&quot;http:&#47;&#47;www.w3.org&#47;2005&#47;Atom&quot;&#41;
+     *     &#47;&#47; - XmlReader.getElementName&#40;&#41;.getLocalPart&#40;&#41;.equals&#40;getRootElementName&#40;rootElementName, &quot;author&quot;&#41;&#41;
+     *     &#47;&#47;
+     *     &#47;&#47; If XmlReader.readObject&#40;String, ReadValueCallback&#41; was used instead, the namespace check would be omitted.
+     *     &#47;&#47;
+     *     &#47;&#47; The ReadValueCallback is where the actual deserialization of the object occurs. When the ReadValueCallback is
+     *     &#47;&#47; called, the XmlReader is positioned at the start of the element that the object is being deserialized from
+     *     &#47;&#47; &#40;in this case the &quot;author&quot; element&#41;.
+     *     return xmlReader.readObject&#40;&quot;http:&#47;&#47;www.w3.org&#47;2005&#47;Atom&quot;, getRootElementName&#40;rootElementName, &quot;author&quot;&#41;,
+     *         reader -&gt; &#123;
+     *             ResponseAuthor author = new ResponseAuthor&#40;&#41;;
+     *
+     *             while &#40;xmlReader.nextElement&#40;&#41; != XmlToken.END_ELEMENT&#41; &#123;
+     *                 QName qName = xmlReader.getElementName&#40;&#41;;
+     *                 String localPart = qName.getLocalPart&#40;&#41;;
+     *                 String namespaceUri = qName.getNamespaceURI&#40;&#41;;
+     *
+     *                 if &#40;&quot;name&quot;.equals&#40;localPart&#41; &amp;&amp; &quot;http:&#47;&#47;www.w3.org&#47;2005&#47;Atom&quot;.equals&#40;namespaceUri&#41;&#41; &#123;
+     *                     author.name = xmlReader.getStringElement&#40;&#41;;
+     *                 &#125;
+     *             &#125;
+     *
+     *             return author;
+     *         &#125;&#41;;
+     * &#125;
+     * </pre>
+     * <!-- end io.clientcore.core.serialization.xml.XmlSerializable.fromXml#XmlReader-String -->
+     *
+     * @param xmlReader The {@link XmlReader} being read.
+     * @param rootElementName Optional root element name to override the default defined by the model. Used to support
+     * cases where the model can deserialize from different root element names.
+     * @param <T> The type of the object.
+     * @return The object that the XML stream represented, may return null.
+     * @throws XMLStreamException If an object fails to be read from the {@code xmlReader}.
+     */
+    static <T extends XmlSerializable<T>> T fromXml(XmlReader xmlReader, String rootElementName)
+        throws XMLStreamException {
+        throw new UnsupportedOperationException("Implementation of XmlSerializable must define this factory method.");
```

Comment:
Why do we need these static methods in the interface? We can't have any default implementation, so, is it required to define these methods here?

Reply:
This was a basic copy from azure-xml. It's similar to the static methods in `JsonSerializable` where there is an expectation that this needs to be overwritten.

Reply:
I don't think this is enforceable for implementation types. So, not sure how adding these static methods will help. Should we also remove the static method in `JsonSerializable` (just in clientcore)?